conda/logic.py

```diff
@@ -344,57 +411,28 @@ def LinearBound_(self, equation, lo, hi, preprocess, polarity):
         else:
             nprune = 0
         # Tighten bounds
+        total = sum(c for c, _ in equation[:nterms])
         if preprocess:
             lo = max([lo, 0])
-            hi = min([hi, sum(c for c, a in equation[:nterms])])
+            hi = min([hi, total])
         if lo > hi:
             return False
         if nterms == 0:
             res = lo == 0
+        elif alg == 'BDD':
+            res = self.BDD_(equation, nterms, lo, hi, polarity)
         else:
```

Comment:
This `else` code path now is also not covered.  I guess this is the "fallback" algorithm?  Should we also pull it out into its own method and write a separate test for it?